data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/JacksonEvent.java

```diff
@@ -272,6 +272,11 @@ public boolean isValueAList(final String key) {
         return node.isArray();
     }
 
+    @Override
+    public Map<Object, Object> getAsMap() {
+        return mapper.convertValue(jsonNode, new TypeReference<Map<Object, Object>>() {});
```

Comment:
May consider define the type reference instance as constant field instead of instantiating new instance at every call.